service/history/workflow/mutable_state_impl.go

```diff
@@ -1317,6 +1291,10 @@ func (e *MutableStateImpl) ClearTransientWorkflowTask() error {
 
 		TaskQueue:             nil,
 		OriginalScheduledTime: timestamp.UnixOrZeroTimePtr(0),
+
+		// QUESTION: should we preserve values here? this is called by nDCBranchMgr
+		SuggestContinueAsNew: false,
+		HistorySizeBytes:     0,
```

Comment:
The callers of UpdateWorkflowTask seems to set all the fields all the time to be explicit, which I like.

My question is: in ReplicateWorkflowTaskScheduledEvent, I preserve these two fields by setting them to the current values, so that they stay the same across retries. But in ReplicateTransientWorkflowTaskScheduled, they can be cleared. Is this one more like the former or the latter? I don't really understand the context that this is called in